docs/tutorials/gallery_adding_components.py

```diff
@@ -0,0 +1,311 @@
+# necessary for your particular assistant. The return type is an [`Iterator`](https://docs.python.org/3/library/stdtypes.html#typeiter) of strings.
+
+# %%
+# !!! note
+#     Ragna also supports streaming responses from the assistant. See the
+#     [example how to use streaming responses](../../generated/examples/gallery_streaming.md)
+#     for more information.
+
+from typing import Iterator
+
+from ragna.core import Assistant, Source
+
+
+class TutorialAssistant(Assistant):
+    def answer(self, prompt: str, sources: list[Source]) -> Iterator[str]:
+        yield (
+            f"This is a default answer. There were {len(sources)} sources."
+            f"The prompt was: "
+            f"{prompt}"
+        )
+
+
+# %%
+# ## Including Custom Python Objects
+
+# %%
+# If the module containing the custom object you want to include is in your
+# [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH),
+# you can either use the [config file](../../references/config.md#referencing-python-objects)
+# to add it, or follow the [Python API](#using-the-python-api-with-custom-objects) instructions below.
+
+# %%
+# If the module containing the custom object you want to include is not in your
+# [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH),
+# suppose it is located at the path `~/tutorials/tutorial.py`. You can add `~/tutorial/` to your
+# [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH) using
+# the command
+#
+# ```bash
+# $ export PYTHONPATH=$PYTHONPATH:~/tutorials/
+# ```
+
+# %%
+# ## Using the Python API with Custom Objects
+
+# We first import some helpers that tell Python where to find our demo document that we will
+# use for RAG
+
+import sys
+from pathlib import Path
+
+sys.path.insert(0, str(Path.cwd().parent))
+
+import documentation_helpers
+
+document_path = documentation_helpers.assets / "ragna.txt"
+
+# %%
+# We next import the [ragna.Rag][] class and set up a chat using the custom objects from above
+
+from ragna import Rag
+
+chat = Rag().chat(
+    documents=[document_path],
+    source_storage=TutorialSourceStorage,
+    assistant=TutorialAssistant,
+)
+
+# %%
+# Before we can ask a question, we need to [`prepare`][ragna.core.Chat.prepare] the chat, which
+# under the hood stores the documents we have selected in the source storage.
+
+_ = await chat.prepare()
+
+# %%
+# Finally, we can get an [`answer`][ragna.core.Chat.answer] to a question.
+
+print(await chat.answer("What is Ragna?"))
+
+# %%
+# ## Using the Web UI with Custom Objects
+
+# %%
+# To include our custom objects in the Web UI, make sure they are in your
+# [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH),
+# and list them in your [configuration file](../../references/config.md).
+
+# %%
+# Here is an example configuration file in the case that you would like to use one:
```

Comment:
We don't need a full configuration file here. We can point to the reference in the documentation and just put the important configuration here. We don't want to drown the useful information in the noise.